botocore/httpchecksum.py

```diff
@@ -250,7 +250,14 @@ def resolve_request_checksum_algorithm(
             supported_algorithms = _SUPPORTED_CHECKSUM_ALGORITHMS
 
         algorithm_name = params[algorithm_member].lower()
-        if algorithm_name not in supported_algorithms:
+        if algorithm_name == "crc32c" and not HAS_CRT:
+            raise FlexibleChecksumError(
+                error_msg=(
+                    "Using CRC32C requires an additional dependency. You will "
+                    "need to pip install botocore[crt] before proceeding."
+                )
+            )
```

Comment:
This should be a MissingDependencyError not a FlexibleChecksumError to stay consistent with our CRT exception messaging. See [here](https://github.com/boto/botocore/blob/develop/botocore/utils.py#L2206-L2210) for an example.

We also want to avoid hardcoding this for one algorithm. We should be able to have a defined list of algorithms supported by the CRT and check against that.

```python
if algorithm_name not in supported_algorithms:
    if not HAS_CRT and algorithm_name in crt_algorithms:
        raise MissingDependencyError(
            error_msg=(
                f"Using {algorithm_name} requires an additional dependency. "
                f"You will need to pip install botocore[crt] before proceeding."
            )
        )  
    raise FlexibleChecksumError(
        error_msg="Unsupported checksum algorithm: %s" % algorithm_name
    )

Reply:
Ok I can open a followup PR. My only question is should `CRC32` be included in `crt_algorithms`? Wondering because we have an implementation with and without the CRT.

Reply:
Everything we support in the CRT can be in the list. We should assert that the dictionary keys we generate [here](https://github.com/boto/botocore/blob/develop/botocore/httpchecksum.py#L466-L470) are all in the list to keep everything in sync.